x-pack/plugins/maps/public/classes/styles/vector/properties/dynamic_orientation_property.js

```diff
@@ -11,10 +11,9 @@ import { VECTOR_STYLES } from '../../../../../common/constants';
 export class DynamicOrientationProperty extends DynamicStyleProperty {
   syncIconRotationWithMb(symbolLayerId, mbMap) {
     if (this._options.field && this._options.field.name) {
-      const targetName = getComputedFieldName(
-        VECTOR_STYLES.ICON_ORIENTATION,
-        this._options.field.name
-      );
+      const targetName = this._field.supportsAutoDomain()
```

Comment:
instead of `this._field.getName()`, use `this.getFieldName()`. `this._field` can be null and `this.getFieldName` already has logic in place to check for null fields.